src-control/Control/RefCount.hs

```diff
@@ -546,6 +547,10 @@ checkForgottenRefs = do
   where
 #endif
 
+-- | Run 'checkForgottenRefs', but ignore the resulting exception, if any.
+ignoreForgottenRefs :: IO ()
+ignoreForgottenRefs = void $ try @_ @SomeException $ checkForgottenRefs
```

Comment:
Shouldn't this be as simple as:
```
writeIORef globalForgottenRef Nothing
```
using `checkForgottenRefs` is just an expensive way of calling `assertNoForgottenRefs` and ignoring its exceptions, and relying on its side effect which is the code above.

Reply:
I thought we should run `assertNoForgottenRefs` so that any forgotten-but-not-yet-checked references are garbage collected. If we don't do this, won't it allow `globalForgottenRef` to be contaminated by references that were created (and forgotten) before `ignoreForgottenRefs`?